pkgs/development/embedded/srec/default.nix

```
@@ -0,0 +1,30 @@
+{ lib, clangStdenv, fetchFromGitHub }:
+let
+  name = "srec";
+  version = "0-unstable-2015-03-06";
+in
+clangStdenv.mkDerivation {
+
+  inherit name version;
+
+  src = fetchFromGitHub {
+    owner = "arkku";
+    repo = "srec";
+    rev = "4bf276fd8abc2b3e57ef0e0b7c44dcacd8d810ee";
+    hash = "sha256-HnQ64CPOA7m4FkMIRI3bx8kySY6uUihVKlfKYqxufSc=";
+  };
+
+  installPhase = ''
+    mkdir -p "$out/bin"
+    cp bin2srec "$out/bin"
+    cp srec2bin "$out/bin"
+  '';
+
+  meta = {
+    description = "Binaries to convert Motorola SREC (S-Record) hex files";
+    homepage = "https://github.com/arkku/srec";
+    license = lib.licenses.mit;
+    platfroms = lib.platforms.linux;
```

```suggestion
    platfroms = lib.platforms.unix;
```

I was able to compile and run the program on `arch64-darwin`. Is there some reason why it should be Linux exclusive?

---

I don't have a way to thest on aarch64 so i put linux only.

---

It seems to me they're working fine, on darwin

```console
$ /nix/store/48sps2wvkcvh7djnyn91kdvb7xi231a9-srec/bin/bin2srec --help
Invalid argument: --help
kk_srec 2015-03-06 - Copyright (c) 2015 Kimmo Kulovesi
Usage: bin2srec [-a <address_offset>] [-o <out.bin>] [-i <in.srec>]
                [-b <bytes_in_address>] [-x <exec_start_address>]
                [-h <header_text>] [-v]
```

```console
$ /nix/store/48sps2wvkcvh7djnyn91kdvb7xi231a9-srec/bin/srec2bin --help
Invalid argument: --help
kk_srec 2015-03-06 - Copyright (c) 2015 Kimmo Kulovesi
Usage: srec2bin ([-a <address_offset>]|[-A]) [-o <out.bin>] [-i <in.srec>] [-v]
```